src/components/NavBar.tsx

```diff
@@ -40,9 +41,27 @@ function NavBar() {
 
   const renderedNavItems = navTitles.map((item) => renderNavItem(item))
 
+  const [prevScrollPos, setPrevScrollPos] = useState(0)
+  const [visible, setVisible] = useState(true)
+
+  const handleScroll = () => {
+    const currentScrollPos = window.pageYOffset
+
+    setVisible(prevScrollPos > currentScrollPos)
+    setPrevScrollPos(currentScrollPos)
+  }
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [prevScrollPos, visible, handleScroll])
+
   return (
     <NavContainer>
-      <MainContainer>{renderedNavItems}</MainContainer>
+      <MainContainer>
+        <div style={{ ...navBarStyles, top: visible ? '0' : '-60px', position: 'fixed' }}>{renderedNavItems}</div>
```

Comment:
consider use styled components instead of inline styling

Reply:
resolved in 5edc1cc